src/main.rs

```diff
@@ -23,14 +24,17 @@ async fn main() -> Result<(), std::io::Error> {
     ))
     .expect("Unexpected error getting a pool");
 
+  let config_clone = config.clone();
+
   HttpServer::new(move || {
     let cors = Cors::permissive();
     App::new()
       .wrap(cors)
       .app_data(web::Data::new(pool.clone()))
+      .app_data(web::Data::new(config.clone()))
       .configure(routes::configure_routes)
   })
-  .bind(("127.0.0.1", config.port()))?
+  .bind(("127.0.0.1", config_clone.port()))?
```

Comment:
jwt secret token must be known in multiple places in the code. I could pass just it, but I think it might be better to pass the whole configuration

Reply:
The idea was to `let config = Config::new()` everytime you need something from the config.

I'm not necessarily against this change (the clonings happen three or four times instead of on every request), I found it weird.

Reply:
I thought about this and while I think it is a viable approach in other languages, right now the way it's setup in rust it would be kinda bad, because it would allocate memory each time to read a setting, and then immediately drops it. Maybe we will refactor it in the future

Reply:
I agree. In Rust apparently allocating memory to later drop it is a crime.